modules/cli/src/main/scala/scala/cli/commands/SetupIde.scala

```diff
@@ -148,14 +149,22 @@ object SetupIde extends ScalaCommand[SetupIdeOptions] {
     val json                      = gson.toJson(details)
     val scalaCliOptionsForBspJson = write(options.shared)
 
+    if (bspJsonDestination.toString.contains(".cache"))
+      value(Left(new FilePermissionsError(
```

Comment:
This is not about file permissions. It's just that Scala CLI doesn't know where to write `.bsp`, so that users should pass an explicit workspace (via `--workspace`, if it's accepted by `setup-ide`, or via `--bsp-directory`).